assets/terraform-modules/bare-metal/flatcar-linux/kubernetes/ssh.tf

```diff
@@ -5,9 +5,7 @@ resource "null_resource" "copy-controller-secrets" {
   # Without depends_on, remote-exec could start and wait for machines before
   # matchbox groups are written, causing a deadlock.
   depends_on = [
-    matchbox_group.install,
-    matchbox_group.controller,
-    matchbox_group.worker,
+    module.controller_profile,
```

Comment:
```suggestion
    module.controller_profile,
]
# Triggered when reprovisioned
triggers = {
    changed = module.controller_profile.id,
}

(delete the ] below)
```
I hope this works (previously it was `null_resource.reprovision-worker-when-ignition-changes.id`)

Reply:
shouldn't the trigger be `module.controller[count.index].clc_config`. `clc_config` is an output produced after being rendered by data source `ct_config` by the controller module.

Doing so would mean that controller ignition data has changed and needs reprovisioning.

Reply:
It could also be a kernel command line setting. I think it makes sense to check if `module.controller_profile` got changed to indicate that a reprovisioing may have happened, that's why I suggested `module.controller_profile.id`.

Reply:
`module.controller_profile.id` is syntactically not correct. Terraform won't accept.

Reply:
Maybe `module.controller_profile[0].id` or the something with `count.index`?

Reply:
Ah, maybe `id` is related to a resource not a module, maybe `module.controller_profile[count.index].null_resource.reprovision-node-when-ignition-changes.id`… or just explicitly the kernel args and console args, too.